framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -203,15 +213,37 @@ public Void visitDeclared(AnnotatedDeclaredType type, Void p) {
          * If any constant-value annotation has &gt; MAX_VALUES number of values provided, treats
          * the value as UnknownVal. Works together with ValueVisitor.visitAnnotation, which issues a
          * warning to the user in this case.
+         *
+         * <p>Additional rules applied for IntVal and IntRange annotations: if anno is IntVal and
+         * the number of values provided is &gt; MAX_VALUES, replace with IntRange; if anno is
+         * IntRange and the range is &le; MAX_VALUES, replace with IntVal
          */
-        private void replaceWithUnknownValIfTooManyValues(AnnotatedTypeMirror atm) {
+        private void replaceWithNewAnnoInSpecialCases(AnnotatedTypeMirror atm) {
             AnnotationMirror anno = atm.getAnnotationInHierarchy(UNKNOWNVAL);
 
             if (anno != null && anno.getElementValues().size() > 0) {
-                List<Object> values =
-                        AnnotationUtils.getElementValueArray(anno, "value", Object.class, false);
-                if (values != null && values.size() > MAX_VALUES) {
-                    atm.replaceAnnotation(UNKNOWNVAL);
+                if (AnnotationUtils.areSameByClass(anno, IntRange.class)) {
+                    Range range = getIntRange(anno);
+                    if (range.to < range.from) {
+                        atm.replaceAnnotation(UNKNOWNVAL);
+                    }
+                } else if (AnnotationUtils.areSameByClass(anno, IntVal.class)) {
+                    List<Long> values =
+                            AnnotationUtils.getElementValueArray(anno, "value", Long.class, true);
+                    if (values != null && values.size() > MAX_VALUES) {
+                        long annoMinVal = Collections.min(new ArrayList<>(values));
```

Comment:
Why is it necessary to create a new ArrayList?  A List is already a Collection, so can't it be passed to `Collections.min`?